hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/CryptoDeleteHandlerTest.java

```diff
@@ -27,169 +27,157 @@
-    private final AccountID transferAccountId =
-            AccountID.newBuilder().accountNum(32134).build();
-    private final Long deleteAccountNum = deleteAccountId.accountNum();
-    private final Long transferAccountNum = transferAccountId.accountNum();
-
-    @Mock
-    private MerkleAccount deleteAccount;
-
-    @Mock
-    private MerkleAccount transferAccount;
-
     private CryptoDeleteHandler subject = new CryptoDeleteHandler();
 
+    @BeforeEach
+    public void setUp() {
+        super.setUp();
+        readableAccounts = emptyReadableAccountStateBuilder()
+                .value(EntityNumVirtualKey.fromLong(accountNum), account)
+                .value(EntityNumVirtualKey.fromLong(deleteAccountNum), deleteAccount)
+                .value(EntityNumVirtualKey.fromLong(transferAccountNum), transferAccount)
+                .build();
+        given(readableStates.<EntityNumVirtualKey, Account>get(ACCOUNTS)).willReturn(readableAccounts);
+        readableStore = new ReadableAccountStore(readableStates);
+    }
+
     @Test
     void preHandlesCryptoDeleteIfNoReceiverSigRequired() throws PreCheckException {
-        final var keyUsed = (JKey) payerKey;
-
-        given(accounts.get(EntityNumVirtualKey.fromLong(deleteAccountNum))).willReturn(deleteAccount);
-        given(accounts.get(EntityNumVirtualKey.fromLong(transferAccountNum))).willReturn(transferAccount);
-        given(deleteAccount.getAccountKey()).willReturn(keyUsed);
-        given(transferAccount.getAccountKey()).willReturn(keyUsed);
-        given(transferAccount.isReceiverSigRequired()).willReturn(false);
+        given(transferAccount.receiverSigRequired()).willReturn(false);
+        given(deleteAccount.key()).willReturn(accountKey);
 
         final var txn = deleteAccountTransaction(deleteAccountId, transferAccountId);
 
-        final var context = new PreHandleContext(store, txn);
+        final var context = new PreHandleContext(readableStore, txn);
         subject.preHandle(context);
 
         assertEquals(txn, context.body());
-        assertEquals(payerKey, context.payerKey());
+        assertEquals(key, context.payerKey());
         basicMetaAssertions(context, 0);
     }
 
     @Test
     void preHandlesCryptoDeleteIfReceiverSigRequiredVanilla() throws PreCheckException {
-        final var keyUsed = (JKey) payerKey;
-
-        given(accounts.get(EntityNumVirtualKey.fromLong(deleteAccountNum))).willReturn(deleteAccount);
-        given(accounts.get(EntityNumVirtualKey.fromLong(transferAccountNum))).willReturn(transferAccount);
-        given(deleteAccount.getAccountKey()).willReturn(keyUsed);
-        given(transferAccount.getAccountKey()).willReturn(keyUsed);
-        given(transferAccount.isReceiverSigRequired()).willReturn(true);
+        given(transferAccount.key()).willReturn(key);
+        given(transferAccount.receiverSigRequired()).willReturn(true);
+        given(deleteAccount.key()).willReturn(accountKey);
```

Comment:
Is there a reason to mock `transferAccount` and `deleteAccount` instead of using real account objects?